services/provider/server/consumer.go

```diff
@@ -225,3 +238,23 @@ func (c *ocsConsumerManager) UpdateConsumerStatus(ctx context.Context, id string
 	klog.Infof("successfully updated Status for StorageConsumer %v", consumerObj.Name)
 	return nil
 }
+
+func getStorageQuotaFromOnboardingTicket(ticket string) (int, error) {
+	ticketArr := strings.Split(ticket, ".")
+	if len(ticketArr) != 2 {
+		return 0, fmt.Errorf("invalid ticket")
+	}
+
+	message, err := base64.StdEncoding.DecodeString(ticketArr[0])
+	if err != nil {
+		return 0, fmt.Errorf("failed to decode onboarding ticket: %v", err)
+	}
+
+	var ticketData services.OnboardingTicket
+	err = json.Unmarshal(message, &ticketData)
+	if err != nil {
+		return 0, fmt.Errorf("failed to unmarshal onboarding ticket %v", err)
+	}
+
+	return int(ticketData.StorageQuotaInGiB), nil
+}
```

Comment:
This will better suit us as a general `parseOnboardingTicket` function, then in the call site you will be able to fetch any piece of information from the ticket. 

You can even move it to the services package and have it as
```go 
ticket := services.DecodeOnboardingTicket(ticketString)
```

Reply:
Make sense, we don't have the util package in the server folder. i will create and add a function.

Reply:
Can we have it here? https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/util/provider.go

This file already deals with concerns regarding onboarding tokens/tickets

Reply:
Sure